### PR TITLE
Ret 1845. Suggest hearing venue.

### PR DESCRIPTION
**Pull request by Itay-Kirsh**
https://tools.hmcts.net/jira/browse/RET-1845

Add suggest hearing venue page.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by Itay-Kirsh** [comment]
Waiting for RET-2033 fix to Bulk tests.

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et1VettingService.java [hunk 1]
```diff
@@ -307,4 +314,24 @@ private void addJurCodeToExistingCollection(CaseData caseData, VettingJurisdicti
         caseData.getJurCodesCollection().add(codesTypeItem);
     }
 
+    public DynamicFixedListType getHearingVenuesList(CaseData caseData) {
+        var dynamicListingVenues = new DynamicFixedListType();
```
Comment (nickwest1) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et1VettingService.java:318]: Avoid using var unless very clear and has good reason to do so

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et1VettingService.java [hunk 2]
```diff
@@ -307,4 +314,24 @@ private void addJurCodeToExistingCollection(CaseData caseData, VettingJurisdicti
...
+
+        dynamicListingVenues.setListItems(new ArrayList<>(
```
Comment (nickwest1) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et1VettingService.java:320]: Can we not use List.of()

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et1VettingService.java [hunk 3]
```diff
@@ -307,4 +314,24 @@ private void addJurCodeToExistingCollection(CaseData caseData, VettingJurisdicti
...
+
+    public String getAddressesHtml(CaseData caseData) {
+        Address claimantAddressUK = caseData.getClaimantType().getClaimantAddressUK();
+        Address claimantWorkAddress = caseData.getClaimantWorkAddress().getClaimantWorkAddress();
+        Address respondentAddress = caseData.getRespondentCollection().get(0).getValue().getRespondentAddress();
+
+        return HEARING_VENUES_TITLE
+            + String.format(CLAIMANT_CONTACT_ADDRESS, toAddressWithTab(claimantAddressUK))
+            + String.format(CLAIMANT_WORK_ADDRESS, toAddressWithTab(claimantWorkAddress))
+            + String.format(RESPONDENT_ADDRESS, toAddressWithTab(respondentAddress));
```
Comment (nickwest1) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et1VettingService.java:335]: Let's just have one template that takes 3 arguments

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/utils/CaseDataBuilder.java [hunk 4]
```diff
@@ -135,7 +136,7 @@ private HearingTypeItem createHearing(String hearingNumber, String hearingType,
 
     public CaseDataBuilder withHearingSession(int hearingIndex, String number, String listedDate, String hearingStatus,
                                               boolean disposed) {
-        var hearing = caseData.getHearingCollection().get(hearingIndex);
+        final var hearing = caseData.getHearingCollection().get(hearingIndex);
 
         var dateListedType = new DateListedType();
```
Comment (nickwest1) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/utils/CaseDataBuilder.java:141]: Where you're changing something that's declared with var, please clean it up

**Review by nickwest1 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et1VettingService.java (hunk 2 above)
Comment (Itay-Kirsh) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et1VettingService.java:320]: Even better, setListItems requires a list and that's what's returned by valueOfOfficeName. No need for a new list.

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/utils/CaseDataBuilder.java (hunk 4 above)
Comment (Itay-Kirsh) [reply src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/utils/CaseDataBuilder.java:141]: I don't quite understand. To clarify my change, this code wasn't mine, I only wanted to improve the checkstyle so I made it constant. I've changed it's position now and made it a type 'val' to remove the checkstyle problem.

**Review by robin-ieong — DISMISSED** [review]

**Review by robin-ieong — DISMISSED** [review]

**Review by robin-ieong — APPROVED** [review]

**Review by nickwest1 — APPROVED** [review]

**Pull request merged**